lib/checks/checks.go

```diff
@@ -822,17 +847,29 @@ func currentServerTime(currentTime, heartbeatTime, serverTime time.Time) time.Ti
 }
 
 func basicCheckers(options *validationpb.ValidateOptions) health.Checker {
-	return monitoring.NewCompositeChecker(
-		"local",
-		[]health.Checker{
-			monitoring.NewIPForwardChecker(),
-			monitoring.NewBridgeNetfilterChecker(),
-			monitoring.NewMayDetachMountsChecker(),
-			monitoring.DefaultProcessChecker(),
-			defaultPortChecker(options),
-			monitoring.DefaultBootConfigParams(),
-		},
-	)
+	checkISCSI := false
+	if options != nil && options.Openebs {
+		checkISCSI = true
+	}
+
+	processesToCheck := DefaultProcessesToCheck
+	if checkISCSI {
+		processesToCheck = append(processesToCheck, "iscsid")
+	}
+
+	checkers := []health.Checker{
+		monitoring.NewIPForwardChecker(),
+		monitoring.NewBridgeNetfilterChecker(),
+		monitoring.NewMayDetachMountsChecker(),
+		monitoring.DefaultProcessChecker(processesToCheck),
+		defaultPortChecker(options),
+		monitoring.DefaultBootConfigParams(),
+	}
+	if checkISCSI {
+		checkers = append(checkers, monitoring.NewISCSIChecker(failedISCSIProbeMessage))
```

Comment:
It made sense to move the process list away from the process checker since it was a detail that was specific to a gravity version but factoring out an error message seems like an overkill? Do you think another error message would be used elsewhere?

Reply:
Moving the error message out of Satellite is needed because the text in the error message contradicts the goal of the Satellite project. Satellite should become a standalone project independent of Gravity. The error message shows "Gravity" and as such if left in Satellite it will make it very specific to Gravity.

The other option is to remove the "Gravity" text from the error message but this contradicts the goal of showing a user friendly error message.  The current error message shown below makes it clear that this is specific to the Gravity application. Embedding a Gravity specific error message into the Satellite project will make it look like they are very tightly coupled. 

Moving the error message out will allow the iscsi checker in Satellite to be used as a general purpose checker. It will make it more independent by allowing possible future usage for other purposes.

For reference below is the Gravity specific message that was moved out of Satellite:

```
failedISCSIProbeMessage = "Found conflicting systemd service: %v. " +
		"If this service is present on the host it will interfere " +
		"with OpenEBS enabled applications running in Gravity." +
		"Please stop and mask this service and try again."

```

Reply:
Ok - my concern is that customizing via an error message is pretty fragile esp. here since it's also a placeholder. It would be more appropriate to move this to a configuration struct or accept a function instead - this way we can at least verify type safety and ensure intent:
```go
monitoring.NewISCSIChecker(fmt FormatISCSIError) Checker { ... }
type FormatISCSIError func(unitName string) string
```